### PR TITLE
2023.07.14 rocprofiler v2 intergration

### PR DESCRIPTION
**Pull request by gcongiu**
## Pull Request Description
Draft PR for rocprofiler V2 interface integration in the rocm component. The PR only implements device wide profiling currently.

## Author Checklist
- [x] **Description**
_Why_ this PR exists. Reference all relevant information, including _background_, _issues_, _test failures_, etc
- [ ] **Commits**
_Commits_ are self contained and only do one thing
_Commits_ have a header of the form: `module: short description`
_Commits_ have a body (whenever relevant) containing a detailed description of the addressed problem and its solution
- [ ] **Tests**
The PR needs to pass all the tests

### CODE REVIEW

**Comment by gcongiu** [comment]
@G-Ragghianti do you know if it is possible to disable GitHub actions for draft PRs?

**Comment by G-Ragghianti** [comment]
> @G-Ragghianti do you know if it is possible to disable GitHub actions for draft PRs?

It looks like it may be possible, but it may make the yaml too complicated to be worth it.  See https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#pull_request

**Comment by gcongiu** [comment]
@G-Ragghianti I would not worry about then

File: src/components/rocm/roc_profiler_v2.c [hunk 1]
```diff
@@ -0,0 +1,798 @@
+#include <rocprofiler.h>
```
Comment (bertwesarg) [inline src/components/rocm/roc_profiler_v2.c:1]: I need this to get it compiling (5.7.0 RC here):

```suggestion
#include <rocprofiler/v2/rocprofiler.h>
```

please also note the usage of the `rocprofiler/` directory. ROCm changed/make it consistent that one should only need one `-I` flag (`/opt/rocm/include`) and use the ROCm component in the `#include` directive. See [here](https://docs.amd.com/en/docs-5.4.0/release.html#wrapper-header-files)

Comment (bertwesarg) [reply src/components/rocm/roc_profiler_v2.c:1]: a full patch could look like this (not tested with V1 and V2):

```diff
diff --git i/src/components/rocm/Rules.rocm.in w/src/components/rocm/Rules.rocm.in
index 0582a0c74..433ab3b03 100644 src/components/rocm/Rules.rocm.in
--- i/src/components/rocm/Rules.rocm.in
+++ w/src/components/rocm/Rules.rocm.in
@@ -20,10 +20,6 @@ endif
 COMPOBJS += rocm.o roc_dispatch.o roc_common.o $(ROCPOBJS)

 CFLAGS += -I$(PAPI_ROCM_ROOT)/include \
-          -I$(PAPI_ROCM_ROOT)/include/hsa \
-          -I$(PAPI_ROCM_ROOT)/hsa/include/hsa \
-          -I$(PAPI_ROCM_ROOT)/include/rocprofiler \
-          -I$(PAPI_ROCM_ROOT)/rocprofiler/include \
           -g

 LDFLAGS += $(LDL)
diff --git i/src/components/rocm/roc_common.h w/src/components/rocm/roc_common.h
index 0cfd95495..743078d2a 100644 src/components/rocm/roc_common.h
--- i/src/components/rocm/roc_common.h
+++ w/src/components/rocm/roc_common.h
@@ -1,7 +1,7 @@
 #ifndef __ROC_COMMON_H__
 #define __ROC_COMMON_H__

-#include <hsa.h>
+#include <hsa/hsa.h>
 #include <dlfcn.h>
 #include <unistd.h>
 #include <stdlib.h>
diff --git i/src/components/rocm/roc_profiler.c w/src/components/rocm/roc_profiler.c
index 78a6e1f0a..0c3b3bfcc 100644 src/components/rocm/roc_profiler.c
--- i/src/components/rocm/roc_profiler.c
+++ w/src/components/rocm/roc_profiler.c
@@ -5,7 +5,7 @@
  *
  */

-#include <rocprofiler.h>
+#include <rocprofiler/rocprofiler.h>
 #include "roc_profiler.h"
 #include "roc_common.h"
 #include "htable.h"
diff --git i/src/components/rocm/roc_profiler_v2.c w/src/components/rocm/roc_profiler_v2.c
index 632e71acc..780020f45 100644 src/components/rocm/roc_profiler_v2.c
--- i/src/components/rocm/roc_profiler_v2.c
+++ w/src/components/rocm/roc_profiler_v2.c
@@ -1,4 +1,4 @@
-#include <rocprofiler.h>
+#include <rocprofiler/v2/rocprofiler.h>
 #include "roc_profiler_v2.h"
 #include "roc_common.h"
 #include "htable.h"

**Comment by bertwesarg** [comment]
I get a segmentation fault when running the `components/rocm/tests/sample_single_kernel_monitoring` with ROCM 5.7.0 RC1:

```
#0  0x00007ffff582ca9f in raise () from /lib64/libc.so.6
#1  0x00007ffff57ffe05 in abort () from /lib64/libc.so.6
#2  0x00007ffff586f037 in __libc_message () from /lib64/libc.so.6
#3  0x00007ffff587619c in malloc_printerr () from /lib64/libc.so.6
#4  0x00007ffff587902c in _int_malloc () from /lib64/libc.so.6
#5  0x00007ffff587a822 in malloc () from /lib64/libc.so.6
#6  0x00007ffff61fa94c in operator new (sz=32) at /dev/shm/GCCcore/11.3.0/system-system/gcc-11.3.0/stage3_obj/x86_64-pc-linux-gnu/libstdc++-v3/libsupc++/new_op.cc:50
#7  0x00007fffebf91409 in rocprofiler_device_profiling_session_create () from /opt/rocm-5.7.0/lib/librocprofiler64.so.2
#8  0x000000000040d357 in open_sampling_ctx (ctx=0xb219c0) at /lustre/ssd/ws/wesarg-amdgpu-metrics/papi-rocprof-v2/src/roc_profiler_v2.c:553
#9  sampling_ctx_open (ctx=0x7fffffffaae8, num_events=<optimized out>, events_id=<optimized out>)
    at /lustre/ssd/ws/wesarg-amdgpu-metrics/papi-rocprof-v2/src/roc_profiler_v2.c:602
#10 rocp2_ctx_open (events_id=<optimized out>, num_events=<optimized out>, ctx=ctx@entry=0x7fffffffaae8)
    at /lustre/ssd/ws/wesarg-amdgpu-metrics/papi-rocprof-v2/src/roc_profiler_v2.c:196
#11 0x000000000040c285 in rocd_ctx_open (events_id=<optimized out>, num_events=<optimized out>, ctx=ctx@entry=0x7fffffffaae8)
    at /lustre/ssd/ws/wesarg-amdgpu-metrics/papi-rocprof-v2/src/roc_dispatch.c:112
#12 0x000000000040c16e in try_open_events (rocm_ctl=0xb206e0) at /lustre/ssd/ws/wesarg-amdgpu-metrics/papi-rocprof-v2/src/rocm.c:418
#13 rocm_update_control_state (ctx=<optimized out>, ntv_count=1, ntv_info=0xbfaa70, ctl=0xb206e0) at /lustre/ssd/ws/wesarg-amdgpu-metrics/papi-rocprof-v2/src/rocm.c:349
#14 rocm_update_control_state (ctl=0xb206e0, ntv_info=0xbfaa70, ntv_count=1, ctx=<optimized out>) at /lustre/ssd/ws/wesarg-amdgpu-metrics/papi-rocprof-v2/src/rocm.c:327
#15 0x00000000003e57b2 in add_native_events (ESI=ESI@entry=0x5b17e0, nevt=nevt@entry=0x7fffffffab7c, size=size@entry=1, out=0xbf3a60) at papi_internal.c:1297
#16 0x00000000003e756a in _papi_hwi_add_event (ESI=0x5b17e0, EventCode=<error reading variable: dwarf2_find_location_expression: Corrupted DWARF expression.>)
    at papi_internal.c:1454
#17 0x00000000003e0869 in PAPI_add_event (EventSet=EventSet@entry=0, EventCode=1073741841) at papi.c:2243
#18 0x00000000003e08d3 in PAPI_add_named_event (EventSet=0, EventName=<optimized out>) at papi.c:2447
#19 0x00000000003de16e in main (argc=1, argv=0x7fffffffae88) at sample_single_kernel_monitoring.cpp:46
```

**Comment by bertwesarg** [comment]
Timour from AMD told me, I should not concentrate on V2 for now, thus will go back to `master` with V1

**Comment by gcongiu** [comment]
Thank you @bertwesarg for your comments. I will look into them in details when I pick this up again.

**Pull request closed without merging**